packages/abi-coder/src/interface.ts

```diff
@@ -4,110 +4,57 @@ import { arrayify } from '@ethersproject/bytes';
 import { Logger } from '@ethersproject/logger';
 import { versions } from '@fuel-ts/versions';
 
-import AbiCoder from './abi-coder';
+import { AbiCoder } from './abi-coder';
 import type { InputValue } from './coders/abstract-coder';
-import type { Fragment } from './fragments/fragment';
-import FunctionFragment from './fragments/function-fragment';
-import type {
-  JsonAbiFragment,
-  JsonFlatAbi,
-  JsonFlatAbiFragmentType,
-  JsonAbi,
-  JsonAbiLogFragment,
-  ConfigurableFragment,
-} from './json-abi';
-import { isFlatJsonAbi, ABI } from './json-abi';
+import { FunctionFragment } from './function-fragment';
+import type { JsonAbi, JsonAbiConfigurable } from './json-abi';
 
 const logger = new Logger(versions.FUELS);
 
-const coerceFragments = (value: ReadonlyArray<JsonAbiFragment>): Array<Fragment> => {
-  const fragments: Array<Fragment> = [];
+export class Interface<TAbi extends JsonAbi = JsonAbi> {
+  readonly functions!: Record<string, FunctionFragment>;
 
-  value.forEach((v) => {
-    if (v.type === 'function') {
-      fragments.push(FunctionFragment.fromObject(v));
-    }
-  });
-
-  return fragments;
-};
-
-export const convertConfigurablesToDict = (value: ReadonlyArray<ConfigurableFragment>) => {
-  const configurables: { [name: string]: ConfigurableFragment } = {};
-
-  value.forEach((v) => {
-    configurables[v.name] = {
-      ...v,
-    };
-  });
-
-  return configurables;
-};
-
-export default class Interface {
-  readonly fragments: Array<Fragment>;
-  readonly functions: { [name: string]: FunctionFragment };
-  readonly configurables: { [name: string]: ConfigurableFragment };
-  readonly abiCoder: AbiCoder;
-  readonly abi: ABI | null;
-  readonly types: ReadonlyArray<JsonFlatAbiFragmentType>;
-  readonly loggedTypes: ReadonlyArray<JsonAbiLogFragment>;
+  readonly configurables: Record<string, JsonAbiConfigurable>;
   /*
-  Same as the `loggedTypes` above, but dedicated to external contracts
-  added via `<base-invocation-scope.ts>.addContracts()` method. This is
-  used to decode logs from contracts other than the main contract
+  TODO: Refactor so that there's no need for externalLoggedTypes
+   
+  This is dedicated to external contracts added via `<base-invocation-scope.ts>.addContracts()` method. 
+  This is used to decode logs from contracts other than the main contract
   we're interacting with.
-*/
-  private externalLoggedTypes: { [id: string]: ReadonlyArray<JsonAbiLogFragment> };
+  */
+  private externalLoggedTypes: Record<string, Interface>;
```

Comment:
This is now a `Record<string, Interface>` because the full json abi is needed to decode a log. The one with the knowledge how to decode a log is the `Interface`, as it encapsulates the whole json abi. If it remained a `Record<string, JsonAbiLoggedtype>`, then the logged type might be of an abi type that the `Interface` it was given to as an `externalLoggedType` cannot decode.